UI QoL Changes

kerrlabajo opened the pull request:
This PR will be responsible for the following implementations:
1. Update training parameters.
2. Add logout.
3. Add help link.
4. Download model should first retrieve list of models from S3 bucket that was fetched from  one or more training jobs. The user will then select the model to download.
5. Add download model progress tracker.
6. Add list of models to download. (This should first check if their bucket have any training jobs and contains any output folder with `model.tar.gz` and `output.tar.gz`.)
7. Add dropdown for hyperparameter options (low, scratch, no augmentation, etc.)
8. Write hyperparameter yaml file to be used for custom settings.

In order to open the task list in your Visual Studio, click `Ctrl+\+T` or search `Task List`. Make sure to remove the task that you have implemented/completed.

kerrlabajo committed:
Add TODO tasks and remove old tasks

This PR will be responsible for the following implementations:
1. Update training parameters.
2. Add logout.
3. Add help link.
4. Download model should first retrieve list of models from S3 bucket that was fetched from  one or more training jobs. The user will then select the model to download.

In order to open the task list in your Visual Studio, click `Ctrl+\+T` or search `Task List`.
Make sure to remove the task that you have implemented.

kerrlabajo committed:
Merge branch `development` that had previous merge from PR #21

Ra-Jay committed:
Add utility class

Ra-Jay committed:
make weights dynamic dependent to image size

Ra-Jay committed:
add utility method GetValueFromTextBox

Ra-Jay committed:
Create a method to create a yaml file based on custom hyperparams

Ra-Jay committed:
Create new for for custom hyperparams input

Ra-Jay committed:
make hyperpams selection a dropdown and open new form on custom

Ra-Jay committed:
change WriteYamlFile return type to string

Ra-Jay committed:
create Help Form

Ra-Jay committed:
open Help Form on Help link click

Ra-Jay committed:
connection initial setup

Ra-Jay committed:
create a helper method that fetch available models

Ra-Jay committed:
fetch available models and display on dropdown selection for download

Ra-Jay commented:
Updates: 

- Add utility class, with the following methods:
     - GetWeightFile
     - GetValueFromTextBox
- Make the selectin of weights dynamic, dependent to the image size selected
- Create CustomHyerperParams form
- Create a method to generate a yaml file containing the user input in the CustomHyperParamsForm
- Add Help page/form (empty as of now)
- Create a method to fetch available models in the S3 bucket to be downloaded

**With fetched models form s3:**
![image](https://github.com/kerrlabajo/LSC-Trainer/assets/58256966/c3771f86-b46b-4d12-ac65-05f8baf93b13)

**Custom Hyperparameter Form: **
![image](https://github.com/kerrlabajo/LSC-Trainer/assets/58256966/8ca977c7-4678-4c86-a6af-57fa1d2967a4)

kerrlabajo committed:
Merge branch `features` from PR #18 and resolve conflicts, and rename most components

kerrlabajo committed:
Update design name to `MainForm` except the file for now

Not yet renaming the file itself because it might remove all prevous commits of each lines due to deleting and adding mechanism of renaming.

kerrlabajo committed:
Set tab index in order for all user related inputs except custom hyperparams

kerrlabajo committed:
Add cursor waiter and `mainPanel` enabler upon finished fetching

kerrlabajo committed:
Instead of cursor, make it only wait within `mainPanel`

kerrlabajo committed:
Return string response for `DownloadObjects`

kerrlabajo committed:
Remove unused components

kerrlabajo committed:
Set default cursor for whenever mouse hovers on menu strip

kerrlabajo committed:
Remove unused components

kerrlabajo committed:
Add wait cursor and set to default after

kerrlabajo committed:
Add loading status, and display to log the response of download

kerrlabajo committed:
Include model list combo box and fetch models button in inputs enabler

kerrlabajo committed:
Add environment new line for when message about dataset deletion is displayed

kerrlabajo commented:
Major merge conflict resolution:
1. Merge branch `features` that had recent push from PR #18 and fix major layout changes.
2. Cherry pick significant components and prune unused items.
3. Generalized naming of labels, and ordering of tab indeces.
4. Set cursor to loading, and disable user inputs except menu strips when a process is running.
![image](https://github.com/kerrlabajo/LSC-Trainer/assets/93184975/e5fdf0eb-e81f-42fd-9493-31b814979f36)

kerrlabajo committed:
Major renaming of `labels`, and `textBox` in CustomHyperParamsForm

kerrlabajo committed:
Add public `HyperParameters` dictionary

kerrlabajo committed:
Reflect all `textBox` naming for all values in the dictionary

kerrlabajo committed:
Declare `customHyperParamsForm`, and set it as `CreateTrainingJobRequest.HyperParamters`

kerrlabajo committed:
Set tab indeces for all `textBox` in order in `CustomHyperparamsForm`

kerrlabajo commented:
Add-ons:
1. Instead of having to create a `.yaml` file locally, there was no way of sending this file virtually, nor able to make it access, unless there was an S3 uploadation (again). Therefore, the only way of doing this is to make use of `CreateTrainingJobRequest.HyperParamters`. However, it will be stored in `/opt/ml/input/config/hyperparameters.json` by default.
2. According to the first add-ons above, this implementation of accessing the `hyperparameters.json` (and possibly convert this into `.yaml` file) would only be held in a different `Pull Request` that involves manipulation of scripting in `training-container/docker/yolov5/train_and_export.py`.
3. The variables should also be aligned similarly to how other hyps like `hyp.no-augmentation.yaml`, `hyp.scratch-low.yaml`, or `hyp.scratch-high.yaml` are structured.
![image](https://github.com/kerrlabajo/LSC-Trainer/assets/93184975/3e2be334-0957-4c8f-a9e5-28a7857a98bd)

After setting the `CreateTrainingJobRequest.HyperParameters` value, it was proven to be working (but not yet accessible nor used in the current training algo image) that the hyperparameters are passed to the running container of SageMaker.
![image](https://github.com/kerrlabajo/LSC-Trainer/assets/93184975/c5739393-f9a8-46b3-bced-7892bd6b5fc5)

kerrlabajo committed:
Fix lowercase labels
